QUALITY_DECLARATION.md

```diff
@@ -0,0 +1,199 @@
+
+There is an automated test which runs a linter (ament_copyright) that ensures each file has a license statement.
+
+### Copyright Statement [3.iv]
+
+The copyright holders each provide a statement of copyright in each source code file in `class_loader`.
+
+There is an automated test which runs a linter (ament_copyright) that ensures each file has at least one copyright statement.
+
+Most recent test results can be found [here](http://build.ros2.org/view/Epr/job/Epr__class_loader__ubuntu_bionic_amd64/lastBuild/testReport/class_loader/copyright_check_tests_only/)
+
+## Testing [4]
+
+### Feature Testing [4.i]
+
+Each feature in `class_loader` has corresponding tests which simulate typical usage, and they are located in the `test` directory.
+
+Currently nightly results can be seen here:
+* [linux-aarch64_release](https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_release/lastBuild/testReport/class_loader/)
+* [linux-arm64_release](https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/class_loader/)
+* [mac_osx_release](https://ci.ros2.org/view/nightly/job/nightly_osx_release/lastBuild/testReport/class_loader/)
+* [windows_release](https://ci.ros2.org/view/nightly/job/nightly_win_rel/lastBuild/testReport/class_loader/)
+
+### Public API Testing [4.ii]
+
+Each part of the public API have tests, and new additions or changes to the public API require tests before being added.
+The tests aim to cover both typical usage and corner cases, but are quantified by contributing to code coverage.
+
+### Coverage [4.iii]
+
+`class_loader` follows the recommendations for ROS Core packages in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#coverage), and opts to use branch coverage instead of line coverage.
+
+This includes:
+
+- tracking and reporting line coverage statistics
+- achieving and maintaining a reasonable branch line coverage (90-100%)
+- no lines are manually skipped in coverage calculations
+
+Changes are required to make a best effort to keep or increase coverage before being accepted, but decreases are allowed if properly justified and accepted by reviewers.
+
+Current coverage statistics can be viewed [here](https://ci.ros2.org/job/ci_linux_coverage/lastBuild/cobertura/).
+
+### Performance [4.iv]
+
+`class_loader` does not currently have performance tests.
+
+### Linters and Static Analysis [4.v]
+
+`class_loader` uses and passes all the standard linters and static analysis tools for a C++ package as described in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#linters-and-static-analysis).
+
+## Dependencies [5]
+
+`class_loader` has run-time and build-time dependencies that need to be considered for this declaration.
+
+### Direct Runtime ROS Dependencies [5.i]/[5.ii]
+
+- console_bridge_vendor
+- rcpputils
```

Comment:
`rcpputils` is [claiming level 4](https://github.com/ros2/rcpputils/pull/47/files#diff-92440392597dfc3b7adb67bc4ed481b3R5), so this package can't be at a lower level than its direct runtime dependencies.

That's why it's important we write the declaration for the lower level packages first.